spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -42,9 +42,12 @@ interface FungibleTokenPacketData {
   amount: uint256
   sender: string
   receiver: string
+  metadata: bytes
```

Comment:
I think it'd be nice to have some reasoning for using bytes instead of string given that I think historically string is used over bytes. This is the only difference I can find between the two:

> As described, a string must use UTF-8 character encoding. A string cannot exceed 2GB.

> As described, bytes can store custom data types, up to 2GB in size.

[source](https://developers.google.com/protocol-buffers/docs/encoding)

Do we anticipate any not UTF-8 character encodings to cause problems? I think we already account for this in ibc-go for [Any's](https://github.com/cosmos/ibc-go/blob/7dcfc2c8e8f3ac35ee2590d6c9e0b2776a46f78a/modules/core/02-client/keeper/events.go#L36) and [packet data](https://github.com/cosmos/ibc-go/blob/7dcfc2c8e8f3ac35ee2590d6c9e0b2776a46f78a/modules/core/04-channel/keeper/events.go#L135) - but probably worth noting for other implementations as this is a subtle issue

Reply:
I don't think so because the metadata is completely used outside of transfer application, any middleware or external user that uses metadata and expects UTF-8 encoding can enforce it on their end.

Otoh, since we are allowing arbitrary data to be parsed on the level above transfer, I don't want to make arbitrary limitations in the transfer level and block a potential usecase

Reply:
In interchain accounts we have a [string `memo` in the `InterchainAccountPacketData`](https://github.com/cosmos/ibc/blob/main/spec/app/ics-027-interchain-accounts/README.md#packet-data). Would it make sense to rename it also to `metadata` and make it `bytes`?